src/nwb_conversion_tools/tools/neo/neo.py

```diff
@@ -0,0 +1,602 @@
+            response_unit = neo_reader.header["signal_channels"]["units"][ei]
+            response_conversion = get_conversion_from_unit(unit=response_unit)
+            response_gain = neo_reader.header["signal_channels"]["gain"][ei]
+            response_name = f"{icephys_experiment_type}-response-{si + 1 + simultaneous_recordings_offset:02}-ch-{ei}"
+
+            response = response_classes[icephys_experiment_type](
+                name=response_name,
+                description=f"Response to: {session_stimulus_type}",
+                electrode=electrode,
+                data=neo_reader.get_analogsignal_chunk(block_index=0, seg_index=si, channel_indexes=ei),
+                starting_time=starting_time,
+                rate=sampling_rate,
+                conversion=response_conversion * response_gain,
+                gain=np.nan,
+            )
+            if icephys_experiment_type != "izero":
+                stim_unit = protocol[2][ei]
+                stim_conversion = get_conversion_from_unit(unit=stim_unit)
+                stimulus = stim_classes[icephys_experiment_type](
+                    name=f"stimulus-{si + 1 + simultaneous_recordings_offset:02}-ch-{ei}",
+                    description=f"Stim type: {session_stimulus_type}",
+                    electrode=electrode,
+                    data=protocol[0][si][ei],
+                    rate=sampling_rate,
+                    starting_time=starting_time,
+                    conversion=stim_conversion,
+                    gain=np.nan,
+                )
+                icephys_recording = nwbfile.add_intracellular_recording(
+                    electrode=electrode, response=response, stimulus=stimulus
+                )
+            else:
+                icephys_recording = nwbfile.add_intracellular_recording(electrode=electrode, response=response)
+
+            recordings.append(icephys_recording)
+
+        sim_rec = nwbfile.add_icephys_simultaneous_recording(recordings=recordings)
+        simultaneous_recordings.append(sim_rec)
+
+    nwbfile.add_icephys_sequential_recording(
+        simultaneous_recordings=simultaneous_recordings, stimulus_type=stimulus_type
+    )
+
+    # TODO
+    # # Add a list of sequential recordings table indices as a repetition
+    # run_index = nwbfile.add_icephys_repetition(
+    #     sequential_recordings=[
+    #         seq_rec,
+    #     ]
+    # )
+
+    # # Add a list of repetition table indices as a experimental condition
+    # nwbfile.add_icephys_experimental_condition(
+    #     repetitions=[
+    #         run_index,
+    #     ]
+    # )
+
+
+def add_all_to_nwbfile(
+    neo_reader,
+    nwbfile=None,
+    use_times: bool = False,
+    metadata: dict = None,
+    write_as: str = "raw",
+    es_key: str = None,
+    write_scaled: bool = False,
+    compression: Optional[str] = "gzip",
+    compression_opts: Optional[int] = None,
+    iterator_type: Optional[str] = None,
+    iterator_opts: Optional[dict] = None,
```

Comment:
Don't need these if we're not using iterative write.

Reply:
This function is actually taking a lot of arguments that are spikeinterface-reminiscent but complete unused here, can you review them and remove the unused?

Reply:
I removed these unused args in the latest commit